### PR TITLE
feat: Introduce CUBEJS_DEV_MODE & improve ENV variables experience

### PR DESCRIPTION
**Pull request by ovr**
Hello!

Thanks

### CODE REVIEW

**Review by hassankhan — APPROVED** [review]

File: docs/Cube.js-Backend/Environment-Variables-Reference.md [hunk 1]
```diff
@@ -15,10 +15,10 @@ behavior. Some of these variables can also be set via [configuration options][li
...
 |------------------------------------|-------------|----------------------------------------------------------------------------------------------------------------------------------------------------------------|---------------------------------------------------------|
 |`CUBEJS_CACHE_AND_QUEUE_DRIVER`     |Caching      |The cache and queue driver to use for the Cube.js deployment. Defaults to `redis`                                                                               |`redis | memory`                                         |
-|`CUBEJS_DEV_MODE`                   |Configuration|If `true`, enabled debug mode. Defaults to `false`.                                                                                                             |`true | false`                                           |
+|`CUBEJS_DEV_MODE`                   |Configuration|If `true`, enables debug mode. Defaults to `false`.                                                                                                             |`true | false`                                           |
```
Comment (keydunov) [inline docs/Cube.js-Backend/Environment-Variables-Reference.md:18]: @hassankhan @ovr why it says 'debug' mode instead of dev mode?

Comment (ovr) [reply docs/Cube.js-Backend/Environment-Variables-Reference.md:18]: @keydunov good catch, my miss..

File: docs/Deployment/Production-Checklist.md [hunk 2]
```diff
@@ -10,7 +10,7 @@ deployment.
 
 ## Enable Production Mode
 
-When running Cube.js in production, make sure `NODE_ENV` is set to `production`.
+When running Cube.js Backend in production make sure `NODE_ENV` is set to `production` and `CUBEJS_DEV_MODE` is not set to `true`.
```
Comment (hassankhan) [inline docs/Deployment/Production-Checklist.md:13]: Can we change the first part of the sentence back to how it was?:

```diff
-When running Cube.js Backend in production
+When running Cube.js in production,
```

Comment (ovr) [reply docs/Deployment/Production-Checklist.md:13]: 👍

**Review by vasilev-alex — APPROVED** [review]

**Pull request merged**